src/Microsoft.AspNet.Routing/VirtualPathData.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Routing
+{
+    /// <summary>
+    /// Represents information about the route and virtual path that are the result of
+    /// generating a URL with the ASP.NET routing middleware.
+    /// </summary>
+    public class VirtualPathData
+    {
+        private string _virtualPath;
+        private readonly IDictionary<string, object> _dataToken;
+
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="VirtualPathData"/> class.
+        /// </summary>
+        /// <param name="router">The object that is used to generate the URL.</param>
+        /// <param name="virtualPath">The generated URL.</param>
+        public VirtualPathData([NotNull] IRouter router, string virtualPath)
+            : this(router, virtualPath, dataTokens: new RouteValueDictionary())
+        {
+        }
+
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="VirtualPathData"/> class.
+        /// </summary>
+        /// <param name="router">The object that is used to generate the URL.</param>
+        /// <param name="virtualPath">The generated URL.</param>
+        /// <param name="dataTokens">The collection of custom values.</param>
+        public VirtualPathData(
+            [NotNull] IRouter router,
+            string virtualPath,
+            IDictionary<string, object> dataTokens)
+        {
+            _dataToken = dataTokens ?? new RouteValueDictionary();
```

Comment:
@Eilon 
Actually, in the codes I showed you on the other day did copy this DataToken in ctor. And In the same codes, I  removed 
https://github.com/aspnet/Routing/blob/fix154/src/Microsoft.AspNet.Routing/Template/TemplateRoute.cs#L224-L230

But, I thought you didn't want that. so I reverted them.

I can do copy in this constructor, but I'm wondering if you still keep the following for statement :
https://github.com/aspnet/Routing/blob/fix154/src/Microsoft.AspNet.Routing/Template/TemplateRoute.cs#L224-L230

I think this is redundant.